lib/script.js

```diff
@@ -182,7 +182,12 @@ function wrapAsyncFunctions(asyncFunctions, sandbox, events, done, sandboxRoot)
             }
           };
         } else {
-          args.push(function() {
+          // CONVENTION:
+          // functions that use promises start with the name 'promisedXXXX', add a callback counter for them
+          if ((/^promised/).test(asyncFunctions[k].name)) {
+            events.emit('addCallback');
+          }
+          args.push(function () {
             if (sandboxRoot._error) return;
             events.emit('finishCallback');
```

Comment:
if the wrap function calls without Promise or callback,
the ```callbackCount``` may be wrong, 
as the number of call to ```addCallback``` and ```finishCallback``` is not in pair?